asmcli/lib/util.sh

```diff
@@ -42,6 +42,28 @@ run_command() {
   return $RETVAL
 }
 
+######
+# check_curl calls curl and returns non-zero if the http code is not 200 or the
+# curl command fails.
+######
+check_curl() {
+  local TMPFILE; TMPFILE=$(mktemp)
```

Comment:
Before we start the process we `cd` into a working directory, it's probably useful to make this a named file and leave it around instead of `mktemp` and `rm`.

Reply:
So if we want to capture the output, the caller of check_curl could also just assign the output to a variable and write it to a file. I think using `check_curl` as a general utility function might be trickier if you need to assign a file for output every time you use it.